### PR TITLE
Rtt 12 budget track other expenses

### PR DESCRIPTION
**Pull request by eduranperez**

### CODE REVIEW

File: app/budget/directives/modals/addExpenseItem/addExpenseItem.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    restrict: 'E',
+    template: require('./addExpenseItem.html'),
+    replace: true,
+    scope: {
+      state: '<',
+      isVisible: '=',
+      expenseItemToEdit: '<?',
+    },
+    link: function (scope) {
+      console.log(scope);
```
Comment (jaroldwong) [inline app/budget/directives/modals/addExpenseItem/addExpenseItem.js:14]: debug code

File: app/reports/budgetComparison/directives/supportAndFunds/supportAndFunds.js [hunk 2]
```diff
@@ -14,9 +14,12 @@ let supportAndFunds = function (StringService) {
       costs: '<',
       funding: '<',
       miscStats: '<',
-      budgetScenarioName: '<'
+      budgetScenarioName: '<',
+      expenses: '<',
+      expenseItemTypes: '<'
     },
     link: function (scope) {
+      console.log(scope);
```
Comment (jaroldwong) [inline app/reports/budgetComparison/directives/supportAndFunds/supportAndFunds.js:22]: debug code

File: app/budget/services/actions/budgetActions.js [hunk 3]
```diff
@@ -427,7 +427,92 @@ class BudgetActions {
...
+					var action = {
+						type: ActionTypes.DELETE_EXPENSE_ITEM,
+						payload: {
+							expenseItemId: expenseItemId
+						}
+					};
+
+					$rootScope.$emit('toast', { message: "Deleted expense", type: "SUCCESS" });
+					BudgetReducers.reduce(action);
+					//BudgetCalculations.calculateExpenseItems(); TODO support if needed
```
Comment (jaroldwong) [inline app/budget/services/actions/budgetActions.js:511]: It looks like deleting item(s) doesn't update the values in the navbar or the Summary table. Is that something we can implement?

I noticed it doesn't do it for the Funds tab and was considering updating that behavior.

File: app/budget/directives/expenseItems/expenseItems.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      scope.addExpenseItem = function (expenseItem) {
+        BudgetActions.createExpenseItem(
+          expenseItem,
+          scope.selectedBudgetScenario.id,
+          expenseItem.message
+        );
+      };
+
+      scope.hideExpenseItem = function (expenseItem) {
+        expenseItem.hidden = true;
```
Comment (jaroldwong) [inline app/budget/directives/expenseItems/expenseItems.js:26]: Does expenseItem have a hidden attribute?

Comment (eduranperez) [reply app/budget/directives/expenseItems/expenseItems.js]: Nope, got rid of those thanks.

File: app/budget/services/actions/budgetActions.js (hunk 3 above)
Comment (eduranperez) [reply app/budget/services/actions/budgetActions.js]: I updated it for the other costs.  Let me know if we want to update the funds on this branch as well.

File: app/budget/directives/budgetSummary/budgetSummary.html [hunk 5]
```diff
@@ -217,8 +235,8 @@
 						<div class="budget-summary__cell-container"></div>
 					</td>
 					<td>
-						<div class="budget-summary__cell-container budget-summary__cell-container--last budget-summary__cell-container--last">
-							{{ toCurrency(selectedBudgetScenario.funds) }}
+						<div class="budget-summary__cell-container budget-summary__cell-container--last ">
+							{{ '-' + toCurrency(selectedBudgetScenario.funds) }}
```
Comment (jaroldwong) [inline app/budget/directives/budgetSummary/budgetSummary.html:239]: This may not make sense. They actually allow for negative fund amounts.

Comment (eduranperez) [reply app/budget/directives/budgetSummary/budgetSummary.html:239]: Ah I see, that makes sense now as to why it was not implemented originally.

File: app/budget/directives/modals/addExpenseItem/addExpenseItem.html [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+									type="button" data-toggle="dropdown"
+									class="btn-type-dropdown">
+						<span class="btn-type-dropdown-label">
+							{{ newExpenseItem.termDescription || 'Select a term'}}
+						</span>
+						<span class="caret"></span>
+					</button>
+
+					<ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
+						<li ng-repeat="term in state.ui.termNav.budgetScenarioDropdownTerms | orderBy:'description'" class="type-item">
```
Comment (jaroldwong) [inline app/budget/directives/modals/addExpenseItem/addExpenseItem.html:18]: I feel like it makes more sense not to order it alphabetical here.

File: app/budget/services/actions/budgetActions.js (hunk 3 above)
Comment (jaroldwong) [reply app/budget/services/actions/budgetActions.js]: Mmm, let's keep it separate. I already made a Jira issue for it.

File: app/budget/directives/modals/addExpenseItem/addExpenseItem.html (hunk 6 above)
Comment (eduranperez) [reply app/budget/directives/modals/addExpenseItem/addExpenseItem.html:18]: Yep, makes sense.

**Review by jaroldwong — APPROVED** [review]

**Pull request merged**